internal/server/utils/response.go

```diff
@@ -16,6 +17,9 @@ type ListResponse[T any] struct {
 
 func ReadJSON(r *http.Request, into any) error {
 	if err := json.NewDecoder(r.Body).Decode(into); err != nil {
+		if errors.Is(err, io.EOF) {
```

Comment:
@ishanarya0 would this error mean the json is empty?

Reply:
Yes. I made the body optional in request.